virtcontainers/container.go

```diff
@@ -1296,8 +1296,89 @@ func (c *Container) updateResources(oldResources, newResources ContainerResource
 		// recalculate the actual number of vCPUs if a different number of vCPUs was removed
 		newResources.VCPUs = oldVCPUs - vcpusRemoved
 	}
+	return nil
+}
+
+func (c *Container) memHotplugValid(mem uint32) (uint32, error) {
+	state, err := c.sandbox.storage.fetchSandboxState(c.sandbox.id)
```

Comment:
sandbox state is already saved in the sandbox structure. Please use that one if available so we can possibly avoid disk IO here.

Reply:
Oh, yes you are right. I will fix it.